### PR TITLE
[12_4_X] Put sampic subdet id behind the era modifier

### PR DESCRIPTION
**Pull request by francescobrivio**
#### PR description:
Backport of #38338 and #38366
This PR puts the totemTimingDetId subdetector behind the era modifier. For Run3 subdet=5, for 2016,2017,2018 subdet=6

#### PR validation:
PR was tested with 136.8562 which was previously failing.

#### Backport:
Backport of #38338 and #38366

FYI @ChrisMisan

### CODE REVIEW

**Comment by francescobrivio** [comment]
backport of #38338

**Comment by francescobrivio** [comment]
test parameters:

 - workflows = 136.8562

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
type ctpps

File: EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py [hunk 1]
```diff
@@ -104,6 +104,7 @@
 totemDAQMappingESSourceXML_TotemTiming = cms.ESSource("TotemDAQMappingESSourceXML",
   verbosity = cms.untracked.uint32(10),
   subSystem = cms.untracked.string("TotemTiming"),
+  sampicSubDetId = cms.untracked.uint32(5),
```
Comment (tvami) [inline EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py:107]:
```suggestion
  sampicSubDetId = cms.uint32(5),
```

Suggestion from @davidlange6 actually

Comment (perrotta) [reply EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py:107]: This makes sense. If applied, it has to be propagated also elsewere in the PR.
And in any case, please fix it in the master first

File: EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py [hunk 2]
```diff
@@ -137,6 +138,7 @@
 from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
 from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
 (ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(ctppsPixelDigis, isRun3 = False )
+(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(totemDAQMappingESSourceXML_TotemTiming, sampicSubDetId = cms.untracked.uint32(6) )
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py:141]:
```suggestion
(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(totemDAQMappingESSourceXML_TotemTiming, sampicSubDetId = 6)
```
(This independently from the parameter being tracked or not. Again, master first)

Comment (francescobrivio) [reply EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py]: addressed in [deb4a01](https://github.com/cms-sw/cmssw/pull/38365/commits/deb4a010c508e4d98efedd1db44d0b917bf7c0c2)

File: EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py (hunk 1 above)
Comment (francescobrivio) [reply EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py]: addressed in [deb4a01](https://github.com/cms-sw/cmssw/pull/38365/commits/deb4a010c508e4d98efedd1db44d0b917bf7c0c2)

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by francescobrivio** [comment]
With the current commits I see this error ([log here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a78335/25519/unitTests/src/Calibration/PPSAlCaRecoProducer/test/testExpressPPSAlCaRecoProducer/testing.log)):
```
----- Begin Fatal Exception 14-Jun-2022 17:27:49 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing ESSource: class=TotemDAQMappingESSourceXML label='totemDAQMappingESSourceXML_TimingDiamond'
Exception Message:
MissingParameter: Parameter 'sampicSubDetId' not found.
----- End Fatal Exception -------------------------------------------------
```

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
There is a failure in reading one of the files,
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a78335/25531/runTheMatrixINPUT-results/1365.0_BuToJpsiK_13+BuToJpsiK_13INPUT+DIGIUP15+RECOUP15+HARVESTUP15/step2_BuToJpsiK_13+BuToJpsiK_13INPUT+DIGIUP15+RECOUP15+HARVESTUP15.log
it's obviously not connected to this change. Do we need to re-trigger tests? To me if this is the only problem, we should just sign it

**Comment by tvami** [comment]
+alca
 * test pass, see the caveat in https://github.com/cms-sw/cmssw/pull/38365#issuecomment-1156229327

**Comment by clacaputo** [comment]
+reconstruction

- backport in line with the original PRs
- test pass, see the caveat in https://github.com/cms-sw/cmssw/pull/38365#issuecomment-1156229327

**Comment by perrotta** [comment]
+1
- For CMSSW_12_4_0

**Pull request merged**